fendermint/vm/resolver/src/pool.rs

```diff
@@ -32,11 +34,12 @@ impl<T> ResolveStatus<T>
 where
     T: Clone + Hash + Eq + PartialEq + Sync + Send + 'static,
 {
-    pub fn new(item: T) -> Self {
+    pub fn new(item: T, use_own_subnet: bool) -> Self {
```

Comment:
Great question, thanks. I am not sure I added proper docs about this anywhere other than the PR readme (checkbox #2). 

So this isn't about `own` and `parent`, it's primarily about `own` and the _children_. 

There are two situations:
1. Someone is relaying a checkpoint to the validators on the parent subnet from one of the many child subnets in the form of a transaction. Say the signature checks out and the relayer has gas, then we are left with a CID. Where do we resolve it from? The providers of the child subnet it came from. Why not ask everyone? Because only the child subnet providers can be expected to have them. Why not ask anyway? Because in Bitswap we send messages that we need a CID, and we receive it back, or a timeout or maybe a Nack. To limit the amount of network use and waste, I only send it to a limited number of peers at a time. The more peers we send it to who _don't_ have it, the longer it will take to find someone who _does_. 
2. The parent validators (_own subnet_ in this context) finalized a block with a transaction to _execute_ a checkpoint. This required the majority of validators to vote that they have it, however, a minority may not have it yet. In this case they want to resolve the CID _now_ otherwise they cannot execute the block, so they should ask those other validators in their own (parent) subnet who voted for it, as well as the child subnet providers - someone should be able to serve it.